xcode/Subconscious/Shared/Parsers/Subtext.swift

```diff
@@ -722,3 +724,21 @@ extension Subtext {
         return shortlinkFor(index: range.lowerBound)
     }
 }
+
+extension Subtext {
+    func block(forParagraph: NSTextParagraph) -> Block? {
+        return blocks.last { b in
+            guard let contentRange = forParagraph.paragraphContentRange else {
+                return false
+            }
+            guard let tcm = forParagraph.textContentManager else {
+                return false
+            }
+            
+            guard let range: Range<String.Index> = Range(NSRange(contentRange, in: tcm), in: base) else {
+                return false
+            }
+            return b.body().range.overlaps(range)
```

Comment:
Let's check that subtext is using the same underlying strings as textContentManager